tokio-channel/src/mpsc/mod.rs

```diff
@@ -939,8 +936,24 @@ impl<T> Drop for Receiver<T> {
     fn drop(&mut self) {
         // Drain the channel of all pending messages
         self.close();
-        while self.next_message().is_ready() {
-            // ...
+
+        loop {
```

Comment:
Is there a reason that the channel only drains on drop (and not if I call `receiver.close()` manually)?

Reply:
The `close` API is to be able to manually drain the messages & process them after in order to not lose any.